src/java/org/apache/cassandra/index/sai/disk/v1/bbtree/BlockBalancedTreeWriter.java

```diff
@@ -488,6 +491,7 @@ private class LeafWriter
         private final Callback callback;
         private final ByteBuffersDataOutput leafOrderIndexOutput = new ByteBuffersDataOutput(2 * 1024);
         private final ByteBuffersDataOutput leafBlockOutput = new ByteBuffersDataOutput(32 * 1024);
+        private final byte[] packedValue = new byte[bytesPerValue];
```

Comment:
This is basically the old scratch buffer, right?

The only minor thing is that `writeCommonPrefix()` and `valueInBounds()` also use the name `packedValue`, so perhaps we could rename this to avoid possible confusion.

Reply:
I have left this as `packedValue` because it is the only packed value. I have removed other usages of `packedValue` in favour of `packedValues`. In some cases, I have removed method parameters where we were passing class-level fields.